presto-cli/src/main/java/io/prestosql/cli/QueryRunner.java

```diff
@@ -160,9 +176,175 @@ private static void setupTokenAuth(
             Optional<String> accessToken)
     {
         if (accessToken.isPresent()) {
-            checkArgument(session.getServer().getScheme().equalsIgnoreCase("https"),
-                    "Authentication using an access token requires HTTPS to be enabled");
+//            checkArgument(session.getServer().getScheme().equalsIgnoreCase("https"),
+//                    "Authentication using an access token requires HTTPS to be enabled");
             clientBuilder.addInterceptor(tokenAuth(accessToken.get()));
         }
     }
+
+    private static void setupOktaAuth(
+            OkHttpClient.Builder clientBuilder,
+            ClientSession session,
+            boolean useOkta) {
+        if (useOkta) {
+            System.out.println("Asking for okta authentication");
+            User user = new User();
+            Server server = new Server(5000);
+            server.setHandler(new AuthenticationHandler(server, user));
+
+            try {
+                server.start();
+
+                // Open browser
+                Desktop desktop = java.awt.Desktop.getDesktop();
+                URI loginUrl = new URI("http://localhost:5000/login");
+                desktop.browse(loginUrl);
+
+                server.join();
+                System.out.println("Server exited");
+                System.out.println("Access Token: " + user.getAccessToken());
+                // TODO: This should set access token
+                setupTokenAuth(clientBuilder, session, Optional.ofNullable(user.getAccessToken()));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+}
+
+class User {
+    String email;
+    String accessToken;
+
+    public String getEmail() {
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    public String getAccessToken() {
+        return accessToken;
+    }
+
+    public void setAccessToken(String accessToken) {
+        this.accessToken = accessToken;
+    }
+}
+
+
+class AuthenticationHandler extends AbstractHandler {
+
+    String REDIRECT_URI = "http://localhost:5000/authorization-code/callback";
+
+    String AUDIENCE = "api://default";
+    String STATE = "SomeState";
```

Comment:
What does this signify?

Reply:
This is something passed to OIDC and while redirecting they pass the param. This is basically used to take user to whichever service they were last (or which action the user was trying to do). We don't need this for our use case. Removing

Reply:
Looks like it's a mandatory param. Bringing back